### PR TITLE
Cleanup to prepare for future canonical score and recommended lints

### PR DESCRIPTION
**Pull request by parlough**
I added the lints found here(https://github.com/dart-lang/linter/tree/master/tool/canonical) not previously present in pedantic to the `analysis_options.yaml` resulting in some minor cleanup to ease the inclusion of those rulesets at some point.

### CODE REVIEW

File: lib/inject/inject_parser.dart [hunk 1]
```diff
@@ -72,7 +72,7 @@ class DartPadInjectException implements Exception {
   final String message;
   DartPadInjectException(this.message);
   @override
-  String toString() => '$message';
```
Comment (domesticmouse) [inline lib/inject/inject_parser.dart:75]: I suspect this was for casting null to the actual string 'null'. I'd hold off changing it until after the null safe conversion

**Review by domesticmouse — APPROVED** [review]

File: lib/inject/inject_parser.dart (hunk 1 above)
Comment (parlough) [reply lib/inject/inject_parser.dart:75]: I've ignored this lint for now to be safe. :)

Comment (domesticmouse) [reply lib/inject/inject_parser.dart:75]: Null safety shakes a lot of these things out into the light of day 😎

**Comment by domesticmouse** [comment]
Thanks for the PR!

**Pull request merged**